libstdc++ fails to build on master

**Describe the bug**
When trying to build libstdc++ from nixpkgs master the build fails with the following error messages:

`/nix/store/0n7bssaiibw6zagg9m2p04994rzjab4m-bash-4.4-p23/bin/bash: -c: line 5: unexpected EOF while looking for matching ``'`
`/nix/store/0n7bssaiibw6zagg9m2p04994rzjab4m-bash-4.4-p23/bin/bash: -c: line 6: syntax error: unexpected end of file`
`make: *** [Makefile:1293: all-ash] Error 1`

I have uploaded the full build log to [Gist](https://gist.github.com/smgb66/5e72272f0d3a404c56ef8b4d2f760606).

**To Reproduce**
Steps to reproduce the behavior:
1. Clone the nixpkgs repository into a folder 
2. Try to build the `libstdcxx5` package from that folder using `nix-build`

**Expected behavior**
The package builds without errors.

**Additional context**
By running git bisect I narrowed the bug down to commit 162a935d3cc. Builds work after checking out the commit right before that.
This issue is unrelated to #41585 which seems to have already been fixed, since it was opened in 2018 and the last successful build was in 2019.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.101, NixOS, 19.09.2036.c49da6435f3 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(root): `"nixos-19.09.2036.c49da6435f3, nixos-unstable-20.03pre211015.8539d5f48f9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: libstdcxx5
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/build-support/cc-wrapper/default.nix
+++ b/pkgs/build-support/cc-wrapper/default.nix
@@ -361,7 +361,11 @@ stdenv.mkDerivation {
       done
     ''
 
+    # There are a few tools (to name one libstdcxx5) which do not work
+    # well with multi line flags, so make the flags single line again
     + ''
+      substituteInPlace $out/nix-support/libc-cflags --replace $'\n' ' '
+
       substituteAll ${./add-flags.sh} $out/nix-support/add-flags.sh
       substituteAll ${./add-hardening.sh} $out/nix-support/add-hardening.sh
       substituteAll ${../wrapper-common/utils.bash} $out/nix-support/utils.bash

